js/components/carousel.js

```diff
@@ -0,0 +1,165 @@
+      this.currentlyActiveIndex - 1
+    ];
+  },
+  get carouselButtons() {
+    return document.querySelectorAll(".carousel-nav-button");
+  },
+  get carouselLength() {
+    return document.querySelectorAll(".product-card-carousel").length;
+  },
+  get carouselCardsContainer() {
+    return document.querySelector(".cards-container");
+  },
+};
+
+//main function to fetch carousel items, sort them, create carousel and its nav buttons below
+export default async () => {
+  try {
+    const data = await fetchCarousel("/carousel");
+    data.sort((a, b) => a.priority - b.priority);
+    createCarouselSection(data);
+    createCarouselNavButtons(data);
+  } catch (err) {
+    console.error(err);
+  }
+};
+
+function createCarouselSection(data) {
+  const carouselItemsMarkup = data
+    .map(
+      ({ link, photoUrl, text }, index) => `
+    <section class="${
+      index === 0 ? "product-card-carousel active" : "product-card-carousel"
+    }">
+        <a href="${link}">
+            <div class="carousel-img-container">
+                <img src="${photoUrl}" alt="product-photo" class="carousel-img">
+                <p class="carousel-data-container">${text}</p>
+            </div>
+        </a>
+    </section>`
+    )
+    .join("");
+
+  state.carouselCardsContainer.innerHTML = `
+      <div class="cards-container">
+      ${carouselItemsMarkup}
+      </div>`;
+}
+
+function createCarouselNavButtons(data) {
+  if (data.length < 6) {
+    const buttonsMarkup = data
+      .map(
+        (el, index) => `
+                <button class="${
+                  index === 0
+                    ? "carousel-nav-button button-active"
+                    : "carousel-nav-button"
+                }" type="button" value="${index}"></button>
+            `
+      )
+      .join("");
+
+    const buttonsContainer = `
+        <div class="coordinates-button-container">
+            ${buttonsMarkup}
+        </div>
+    `;
+
+    carousel.insertAdjacentHTML("beforeend", buttonsContainer);
+    addEventListeners(state.carouselButtons);
+  }
+}
+
+function addEventListeners(elements) {
+  elements.forEach((btn) => btn.addEventListener("click", updateSlideButtons));
+}
+
+//below - event listeners for previous/next buttons
+nextButton.addEventListener("click", () => {
+  const isLast = state.currentlyActiveIndex === state.carouselLength - 1;
+
+  if (!isLast) {
+    const offset = -state.currentSlideElement.offsetWidth;
+    updateState(offset, "nextSlideElement", "nextSlideButton");
+    state.currentlyActiveIndex++;
+  }
+});
+
+previousButton.addEventListener("click", () => {
+  const isFirst = state.currentlyActiveIndex === 0;
+
+  if (!isFirst) {
+    const offset = state.currentSlideElement.offsetWidth;
+    updateState(offset, "previousSlideElement", "previousSlideButton");
+    state.currentlyActiveIndex--;
+  }
+});
+
+//updating state depending on the value of buttons above
+function updateState(offset, targetSlide, targetButton) {
```

Comment:
I recommend making `targetSlide` an element, so that you can call it like so:

```
updateState(offset, state.previousSlideElement, state.previousSlideButton)
```


Also, no other values are used for `targetSlide` or `targetButton`. We could simplify and maybe this would work better as a method in `state`

```
state = {
  update(offset) { ... }
}
```